consensus/consensus-types/src/block.rs

```diff
@@ -264,6 +266,26 @@ impl Block {
                 "Reconfiguration suffix should not carry payload"
             );
         }
+        if let Some(failed_authors) = self.block_data().failed_authors() {
+            // when validating for being well formed,
+            // allow for missing failed authors,
+            // for whatever reason (from different max configuration, etc),
+            // but don't allow anything that shouldn't be there.
+            let expected_length = self.round() - parent.round();
+            ensure!(
+                failed_authors.len() <= expected_length as usize,
+                "Block has more failed authors than missed rounds"
+            );
+            let mut bound = parent.round();
+            for (round, _) in failed_authors {
```

Comment:
we probably can check it with a rev().iter().fold(self.round(), |next_round, current_round| ensure!(current_round + 1 = next_round); current_round) to ensure it's contiguous from self.round

Reply:
I am leaving this as a "well formed" check, so it doesn't depend on the rest of the code and configuration, and have actual correctness check in round_manager.